### PR TITLE
Fixed #2098 Error Standardizing

### LINKED ISSUE #2098: Standardize Error Codes

### PR DESCRIPTION
**Pull request by pasin**
* Instead of mapping network domain errors to CFNetwork errors, mapping them to CBL domain errors with the base error of 5000.
* Not mapping POSIX network error-> keep them in POSIX domains.
* Added CBLAssertNotNil to do null check.
* Updated error messages

### CODE REVIEW

File: Objective-C/CBLDatabase.mm [hunk 1]
```diff
@@ -302,14 +312,14 @@ - (BOOL) delete: (NSError**)outError {
         [self mustBeOpen];
         
         if (_activeReplications.count > 0) {
-            NSString* err = @"Cannot delete the database. Please stop all of "
-            "the replicators before closing the database.";
+            NSString* err = @"Cannot delete the database. "
+                "Please stop all of the replicators before closing the database.";
```
Comment (hideki) [inline Objective-C/CBLDatabase.mm]: this is delete() method. message is not correct

File: Objective-C/CBLDatabase.mm [hunk 2]
```diff
@@ -302,14 +312,14 @@ - (BOOL) delete: (NSError**)outError {
...
             return createError(CBLErrorBusy, err, outError);
         }
         
         if (_liveQueries.count > 0) {
-            NSString* err = @"Cannot delete the database. Please remove all of "
-            "the query listeners before deleting the database";
+            NSString* err = @"Cannot delete the database. "
+                "Please remove all of the query listeners before closing the database.";
```
Comment (hideki) [inline Objective-C/CBLDatabase.mm]: this is delete() method. message is not correct

File: Objective-C/Internal/CBLData.mm [hunk 3]
```diff
@@ -37,8 +37,9 @@ - (id) cbl_toPlainObject {
 - (id) cbl_toCBLObject {
     if (self != kCBLRemovedValue) {
         [NSException raise: NSInternalInconsistencyException
-                    format: @"Instances of %@ cannot be added to Couchbase Lite documents",
-                             [self class]];
+                    format: @"%@ is not a valid type. You may only pass NSNumber, "
+                             "NSString, NSDate, CBLBlob, CBLDictionary, CBLArray or "
+                             "NSDictionary/NSArray containing the above types.", [self class]];
```
Comment (hideki) [inline Objective-C/Internal/CBLData.mm]: No Boolean?

Comment (pasin) [reply Objective-C/Internal/CBLData.mm]: Boolean is NSNumber.

File: Objective-C/CBLDatabase.mm (hunk 2 above)
Comment (pasin) [reply Objective-C/CBLDatabase.mm]: Thanks for catching this.

File: Objective-C/CBLDatabase.mm (hunk 1 above)
Comment (pasin) [reply Objective-C/CBLDatabase.mm]: Thanks for catching this.

**Comment by pasin** [comment]
I push the fix for error messages in Database.delete(). Can you review? Thank you very much.

**Pull request merged**